wisp-tracing/src/main/kotlin/wisp/tracing/TracerExt.kt

```diff
@@ -1,38 +1,51 @@
 package wisp.tracing
 
+import io.opentracing.Scope
 import io.opentracing.Span
 import io.opentracing.Tracer
 import io.opentracing.tag.Tags
 
-/** [trace] traces the given function with the specific span name and optional tags */
+/**
+ * Traces a function [f], using a span called [spanName], which is automatically finished when the
+ * function completes execution.
+ *
+ * If a span is already active, the new span is made a child of the existing one.
+ * If you want to manipulate the [Span] (e.g. to attach baggage), use [traceWithSpan] instead.
+ *
+ * If you want a new independent span, use [traceWithNewRootSpan].
+ */
 fun <T : Any?> Tracer.trace(spanName: String, tags: Map<String, String> = mapOf(), f: () -> T): T =
     traceWithSpan(spanName, tags) { f() }
 
-/** [traceWithSpan] traces the given function, passing the span into the function.
- *  If a span is already active, the new span is made a child of the existing. */
+/**
+ * Like [trace], but exposes the new active [Span] to [f].
+ */
 fun <T : Any?> Tracer.traceWithSpan(
     spanName: String,
     tags: Map<String, String> = mapOf(),
     f: (Span) -> T
-): T {
-    return traceWithSpanInternal(spanName, tags, true, f)
-}
+): T = traceWithSpanInternal(spanName, tags, asChild = true, retainBaggage = false, f)
```

Comment:
why are we setting `retainBaggage = false` ?

Reply:
Oh 🤦 it doesn't actually do anything for this method. i.e. baggage retention setting is only applied on new root traces. true/false here doesn't matter. I suppose I should refactor this to avoid this confusion!

Reply:
Fixed!